src/lib.rs

```diff
@@ -92,16 +94,20 @@ pub fn reflink_or_copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> io::Resu
         tracing_attributes::instrument(name = "reflink_or_copy")
     )]
     fn inner(from: &Path, to: &Path) -> io::Result<Option<u64>> {
-        if !from.is_file() {
-            Err(io::Error::new(
-                io::ErrorKind::InvalidInput,
-                "the source path is not an existing regular file",
-            ))
-        } else if let Err(_err) = sys::reflink(from, to) {
+        if let Err(_err) = sys::reflink(from, to) {
             #[cfg(feature = "tracing")]
             tracing::warn!(?_err, "Failed to reflink, fallback to fs::copy");
 
-            fs::copy(from, to).map(Some)
+            fs::copy(from, to).map(Some).map_err(|err| {
```

Comment:
Ideally, I would want to avoid the logging and `fs::copy` if `from` is not a file, however the [`io::ErrorKind::IsADirectory`](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.IsADirectory) is still unstable and it alone is not enough to detect not-a-file situation.

Reply:
would you like me to lift the is_file check? Honestly, I don't care that much about `reflink_or_copy` per se because anyone using it is already consenting to some potential overhead.

Reply:
Well, I think the current code is OK here and doesn't need more change.

It's just unfortunate that `io::ErrorKind` does not contain enough information for us to decide this without having another `.is_file()` call